crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -85,51 +121,127 @@ pub(crate) struct TransactionBlockEffects {
     pub gas_effects: GasEffects,
     pub status: ExecutionStatus,
     pub errors: Option<String>,
-    // pub transaction_block: TransactionBlock,
-    // pub dependencies: Vec<TransactionBlock>,
-    // pub lamport_version: Option<u64>,
+
+    #[graphql(skip)]
+    pub tx_block_digest: Digest,
+    // pub transaction_block: Option<Box<TransactionBlock>>,
+    #[graphql(skip)]
+    pub dependencies: Vec<TransactionDigest>,
+    pub lamport_version: Option<u64>,
+    // unclear what object reads is about, TODO @ashok
     // pub object_reads: Vec<Object>,
     // pub object_changes: Vec<ObjectChange>,
-    // pub balance_changes: Vec<BalanceChange>,
-    // pub epoch: Epoch
-    // pub checkpoint: Checkpoint
+    pub balance_changes: Option<Vec<Option<BalanceChange>>>,
+    // have their own resolvers in the impl block
+    #[graphql(skip)]
+    pub epoch_id: u64,
+    // pub epoch: Option<Epoch>,
+    // pub checkpoint: Option<Checkpoint>,
 }
 
-impl From<&SuiTransactionBlockEffects> for TransactionBlockEffects {
-    fn from(tx_effects: &SuiTransactionBlockEffects) -> Self {
+impl TransactionBlockEffects {
+    pub fn from_stored_transaction(
+        balance_changes: Vec<Option<Vec<u8>>>,
+        _object_changes: Vec<Option<Vec<u8>>>,
+        tx_effects: &SuiTransactionBlockEffects,
+        tx_block_digest: Digest,
+    ) -> Option<Self> {
         let (status, errors) = match tx_effects.status() {
             SuiExecutionStatus::Success => (ExecutionStatus::Success, None),
             SuiExecutionStatus::Failure { error } => {
                 (ExecutionStatus::Failure, Some(error.clone()))
             }
         };
+        let lamport_version = tx_effects
+            .created()
+            .first()
+            .map(|x| x.reference.version.value());
 
-        Self {
+        let balance_changes = balance_changes
+            .into_iter()
+            .map(|x| {
+                x.map(|y| {
+                    let balance_change: NativeBalanceChange = bcs::from_bytes(&y).unwrap();
+                    let owner = Owner {
+                        address: SuiAddress::from_bytes(
+                            balance_change.owner.get_owner_address().unwrap().to_vec(),
+                        )
+                        .unwrap(),
+                    };
+                    let amount =
+                        BigInt::from_str(balance_change.amount.to_string().as_str()).unwrap();
+                    BalanceChange {
+                        owner: Some(owner),
+                        amount: Some(amount),
+                    }
+                })
+            })
+            .collect::<Vec<_>>();
+
+        Some(Self {
             // TODO (wlmyng): To remove as this is the wrong digest, effects digest is not a field on SuiTransactionBlockEffects
```

Comment:
since you're touching this code, would you mind completing this todo and strip digest off TransactionBlockEffects?

Reply:
Like completely? From the Schema too?

Reply:
Yes, we removed the digest field from draft_target_schema.graphql